sentry-apollo/src/main/java/io/sentry/apollo/SentryApolloInterceptor.kt

```diff
@@ -0,0 +1,105 @@
+package io.sentry.apollo
+
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Query
+import com.apollographql.apollo.api.Subscription
+import com.apollographql.apollo.exception.ApolloException
+import com.apollographql.apollo.interceptor.ApolloInterceptor
+import com.apollographql.apollo.interceptor.ApolloInterceptor.CallBack
+import com.apollographql.apollo.interceptor.ApolloInterceptor.FetchSourceType
+import com.apollographql.apollo.interceptor.ApolloInterceptor.InterceptorRequest
+import com.apollographql.apollo.interceptor.ApolloInterceptor.InterceptorResponse
+import com.apollographql.apollo.interceptor.ApolloInterceptorChain
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import java.util.concurrent.Executor
+
+class SentryApolloInterceptor(
+    private val hub: IHub = HubAdapter.getInstance(),
+    private val beforeSpan: BeforeSpanCallback? = null
+) : ApolloInterceptor {
+
+    constructor(hub: IHub) : this(hub, null)
+    constructor(beforeSpan: BeforeSpanCallback) : this(HubAdapter.getInstance(), beforeSpan)
+
+    override fun interceptAsync(request: InterceptorRequest, chain: ApolloInterceptorChain, dispatcher: Executor, callBack: CallBack) {
+        val activeSpan = hub.span
+        if (activeSpan == null) {
+            chain.proceedAsync(request, dispatcher, callBack)
+        } else {
+            val span = startChild(request, activeSpan)
+            val sentryTraceHeader = span.toSentryTrace()
+
+            // we have no access to URI, no way to verify tracing origins
```

Comment:
Should we address this before we ship it?
Sounds like we should bring tracing options (allowlist of outgoing URLs to add the header info) going forward

Reply:
looks like a limitation of apollo

Reply:
you can do with the response object, `response.httpResponse.get().request().url()` but its too late